File: src/main/java/uz/com/journal/app/marks/MarkService.java
package uz.com.journal.app.marks;


import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import uz.com.journal.core.NonDeletableService;
import uz.com.platform.app.users.User;

public interface MarkService extends NonDeletableService<Mark> {
    Page<Mark> findPage(MarkFetch fetch, MarkCriteria criteria, Pageable pageable, User currentUser);

    Mark findOne(String uuid, MarkFetch fetch, User currentUser);

    Mark create(MarkItem item, User currentUser);

    Mark update(String uuid, MarkItem item, User currentUser);
}

File: src/main/webapp/WEB-INF/src/js/app/administration/marks/MarkService.js
module.exports = function ($http) {
    var _this = this;

    _this.findPage = function () {
        var params = {
            size: 10000000,
            page: 0
        };
        return $http.get('/api/marks', {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.findOne = function (uuid) {
        var params = {};
        return $http.get('/api/marks/' + uuid, {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.create = function (subject) {
        return $http.post('/api/marks/', subject).then(function (response) {
            return response.data;
        });
    };

    _this.update = function (subject) {
        return $http.put('/api/marks/' + subject.uuid, subject).then(function (response) {
            return response.data;
        })
    };

    _this.delete = function (subject) {
        return $http.delete('/api/marks/' + subject.uuid);
    }
};